### DISCUSSION TITLE
Optional vs Required Parentheses in Control Flow

### DISCUSSION THREAD

**Discussion by msadeqhe** [carbon-language/carbon-lang#3490] [Polls]
Functions require parentheses. Control flows can have optional parentheses while braces are required:

```cpp
// Optional Parentheses          | // Required Parentheses
if value < 10 {                  | if (value < 10) {
    //statements...              |     //statements...
}                                | }
else if value == 10 {            | else if (value == 10) {
    //statements...              |     //statements...
}                                | }
                                 |
for (var n: i32) in numbers {    | for (var n: i32 in numbers) {
    //statements...              |     //statements...
}                                | }
                                 |
while i < 10 {                   | while (i < 10) {
    //statements...              |     //statements...
    ++i;                         |     ++i;
}                                | }
```

Comparing these two alternative syntaxes...

**Comment by jonmeow** [comment]
Leads evaluated optional parentheses as part of #623. Rationale for requiring parentheses may be found [in that proposal](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0623.md#optional-parentheses).

As a general note regarding polls, bear in mind that there's a relatively small number of people active in design discussions, but a very large target pool of developers. This GitHub forum doesn't provide a representative vote from developers, so polls will have [very low accuracy and] very limited influence on decisions. Providing objective data or new information will typically be a better way to shift a decision.
